pkg/server/image_pull.go

```diff
@@ -255,15 +256,20 @@ func (c *criService) getResolver(ctx context.Context, ref string, cred func(stri
 	if err != nil {
 		return nil, imagespec.Descriptor{}, errors.Wrap(err, "parse image reference")
 	}
+	client := &http.Client{CheckRedirect: checkRedirecter(ctx)}
+	if c.config.Registry.DisableHTTP2 {
+		client.Transport = http.DefaultTransport.(*http.Transport).Clone()
```

Comment:
I think it's weird to do this type cast twice. Can you just make a temporary `transport` variable that receives the clone, assign to its `TLSNextProto`, then set `client.Transport = transport`?

Reply:
And I thought it would be weird to make a temp transport hahaha I'll change

Reply:
Done